src/dns.rs

```diff
@@ -10,6 +11,10 @@ pub trait ToIpAddr {
     fn to_ip_addr(&self, dns: &mut Dns) -> IpAddr;
 }
 
+pub trait ToIpAddrs {
```

Comment:
If we break the contract and need to bump the version anyways, should we consider consolidating into a single trait?

Reply:
We could do, however there is the need for a trait like `ToIpAddr` for the `sim#client` and `sim#host` functions. As in these cases, the trait doubles as a generator rather than only a matcher.